### ISSUE TITLE
ReturnFormat=array doesn't work w/Paginate

### ISSUE THREAD

**Issue by JamoCA** [coldbox-modules/qb#248]
Using qb v9.02 with Adobe ColdFusion 2021u5, this works when using returnFormat "query":
```java
queryBuilder = new qb.models.Query.QueryBuilder(argumentcollection=makeQbInit(DB="SartData", returnFormat="query"));
q = queryBuilder.from("Clients").select("ID, Company");
qry = q.paginate(1, 50);
```

but it throws an error when only changing returnFormat to "array" (and it previously worked in qb v8):
```java
queryBuilder = new qb.models.Query.QueryBuilder(argumentcollection=makeQbInit(DB="SartData", returnFormat="array"));
q = queryBuilder.from("Clients").select("ID, Company");
qry = q.paginate(1, 50);
```

The error message is:
```
You have attempted to dereference a scalar variable of type class java.util.ArrayList as a structure with members.
 The error occurred in qb/models/Query/QueryBuilder.cfc: line 3171
var result = get( options = options );
3171 : if ( result.recordCount <= 0 && !isNull( defaultValue ) ) {
3172 :     return defaultValue;
3173 : } else {
```

**Comment by JamoCA** [comment]
I have been able to manually work around this w/ACF2021 by taking the `results` key from the paginate method and performing a QoQ with a return type of "array".  It's dirty, but it works.

```java
data = q.paginate(1, 50);
data.results = queryexecute("SELECT * FROM [data].results", {}, {"dbtype":"query", "returntype":"array"});
```

**Comment by elpete** [comment]
This should be fixed in qb 9.2.1
